core/src/main/scala/org/apache/spark/sql/SnappyParser.scala

```diff
@@ -699,6 +702,16 @@ class SnappyParser(session: SnappySession)
     )
   }
 
+  protected final def relationPut: Rule1[LogicalPlan] = rule {
+    relationFactor ~ (
+        join. + ~> ((r1: LogicalPlan, joins: Any) => joins.asInstanceOf[
```

Comment:
Cannot be this part and parts from relation (line 697) be a common rule? Just in case, though heard you saying difficulty in code refactoring.